rest-api/test/unit_test/dao_test/participant_summary_dao_test.py

```diff
@@ -375,6 +375,56 @@ def testUpdateEnrollmentStatus(self):
     self.assertEquals(ehr_consent_time, summary.enrollmentStatusMemberTime)
 
 
+  def testUpdateEnrollmentStatusLastModified(self):
+    """
+    DA-631: enrollment_status update should update last_modified.
+    """
+    participant = self._insert(Participant(participantId=6, biobankId=66))
+    self.assertEquals(self.dao.get(participant.participantId).numBaselineSamplesArrived, 0)
+
+    ## Test Step 1: Validate that update_enrollment_status() changes the lastModified value.
+
+    # collect current modified and enrollment status
+    summary = self.dao.get(participant.participantId)
+    self.assertEquals(EnrollmentStatus.INTERESTED, summary.enrollmentStatus)
+    last_modified = summary.lastModified
+
+    # sleep 1 sec to make lastModified different
+    time.sleep(1)
+
+    tmp_summary = ParticipantSummary(
+      participantId=1,
+      biobankId=2,
+      consentForStudyEnrollment=QuestionnaireStatus.SUBMITTED,
+      consentForElectronicHealthRecords=QuestionnaireStatus.SUBMITTED,
+      consentForElectronicHealthRecordsTime=datetime.datetime(2018, 3, 1),
+      enrollmentStatus=EnrollmentStatus.INTERESTED)
+
+    self.dao.update_enrollment_status(tmp_summary)
+    self.assertEquals(EnrollmentStatus.MEMBER, tmp_summary.enrollmentStatus)
+    self.assertNotEqual(last_modified, tmp_summary.lastModified)
```

Comment:
Why do we need to test that two different participants have different last_modified timestamps ?

Reply:
Its testing two different code paths that can change the enrollment status that are independent of each other.

Reply:
Good catch, I wasn't quite getting what you were saying.  I've changed the tmp_summary to use the last_modified value to insure that the datetime assert is valid.  I'm committing the change.

Reply:
I guess what's confusing me here is the comparison between 2 participants. We could use FakeClock to set an exact time for last_modified and test that it changes to the new time.
[https://github.com/all-of-us/raw-data-repository/blob/master/rest-api/test/unit_test/api_test/participant_summary_api_test.py#L252](url)

Reply:
The code path update_from_biobank_stored_samples() updates the data store, so it needs a real participant record for the test to work.  The second code path update_enrollment_status() only updates the summary object's properties.  The code is really about making sure that the summary enrollment changes for the two code paths.  I have committed a new change that only uses one participant for both code path tests.  Let me know if that is what you are looking for.